service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/UploadAdditionalApplicationsSubmittedControllerTest.java

```diff
@@ -586,6 +589,37 @@ void shouldSendNotificationsWhenTriggerEventFails() {
             any()));
     }
 
+
+    @Test
+    @SuppressWarnings("unchecked")
+    void shouldConvertC2ConfidentialBundle() {
+        given(uploadAdditionalApplicationsService.getApplicationTypes(any()))
+            .willReturn(List.of(ApplicationType.C2_APPLICATION));
+
+        C2DocumentBundle c2 = C2DocumentBundle.builder()
+            .type(WITH_NOTICE)
+            .supplementsBundle(new ArrayList<>())
+            .applicantName(LOCAL_AUTHORITY_1_NAME + ", Applicant")
+            .build();
+
+        CaseDetails caseDetails = createCase(ImmutableMap.<String, Object>builder()
+            .putAll(buildCommonNotificationParameters())
+            .put("sendToCtsc", NO)
+            .put("additionalApplicationType", List.of(C2_ORDER))
+            .put("additionalApplicationsBundle", wrapElementsWithUUIDs(AdditionalApplicationsBundle.builder()
+                .pbaPayment(PBAPayment.builder().usePbaPayment(NO.getValue()).build())
+                .c2DocumentBundleConfidential(c2)
```

Comment:
add test cases for `getC2DocumentBundle` and `getOtherApplicationsBundle` as well?

Reply:
added